src/cpp/shared/combat/army.cpp

```diff
@@ -695,6 +822,101 @@ int army::ValidPath(int hex, int flag) {
   return 0;
 }
 
+int army::ValidFlightPath(int hex, int flag)
```

Comment:
How does this function differ from `ValidFlight`?  It might be worth commenting what's changed and why it's needed.